example/methods/finite_difference/finite_difference.cpp

```diff
@@ -0,0 +1,194 @@
+    T_halo->gather( exec_space(), *T );
+
+    // create and array to store previous temperature for explicit udpate
+    auto T_prev = Cajita::createArray<double, device_type>( name, layout );
+    auto T_prev_view = T_prev->view();
+
+    // Solve heat conduction from point source
+    double dt = 1e-6;
+    double end_time = 1e-3;
+    int num_steps = static_cast<int>( end_time / dt );
+
+    // Material properties
+    double rho = 1000.0;
+    double specific_heat = 1000.0;
+    double kappa = 10.0;
+
+    double alpha = kappa / ( rho * specific_heat );
+    double alpha_dt_dx2 = alpha * dt / ( cell_size * cell_size );
+    double dt_rho_cp = dt / ( rho * specific_heat );
+
+    // gaussian heat source parameters (sigma is std dev of gaussian)
+    double eta = 0.1;
+    double power = 100.0;
+    double sigma[3] = { 50e-6, 50e-6, 25e-6 };
+    double r[3] = { sigma[0] / sqrt( 2.0 ), sigma[1] / sqrt( 2.0 ),
+                    sigma[2] / sqrt( 2.0 ) };
+
+    double I = 2.0 * eta * power / ( M_PI * sqrt( M_PI ) * r[0] * r[1] * r[2] );
+
+    // Timestep loop.
+    for ( int step = 0; step < num_steps; ++step )
+    {
+        // Store previous value for explicit update
+        Kokkos::deep_copy( T_prev_view, T_view );
+
+        // Solve finite difference
+        Cajita::grid_parallel_for(
+            "finite_difference", exec_space(), owned_space,
+            KOKKOS_LAMBDA( const int i, const int j, const int k ) {
+                double loc[3];
+                int idx[3] = { i, j, k };
+                local_mesh.coordinates( Cajita::Cell(), idx, loc );
+
+                double f = ( loc[0] * loc[0] / r[0] / r[0] ) +
+                           ( loc[1] * loc[1] / r[1] / r[1] ) +
+                           ( loc[2] * loc[2] / r[2] / r[2] );
+
+                double Q = I * exp( -f ) * dt_rho_cp;
+
+                double laplacian = ( -6.0 * T_prev_view( i, j, k, 0 ) +
+                                     T_prev_view( i - 1, j, k, 0 ) +
+                                     T_prev_view( i + 1, j, k, 0 ) +
+                                     T_prev_view( i, j - 1, k, 0 ) +
+                                     T_prev_view( i, j + 1, k, 0 ) +
+                                     T_prev_view( i, j, k - 1, 0 ) +
+                                     T_prev_view( i, j, k + 1, 0 ) ) *
+                                   alpha_dt_dx2;
+
+                T_view( i, j, k, 0 ) += laplacian + Q;
+            } );
+
+        // Update the boundary conditions
+        updateBoundaries( exec_space(), local_grid, T_view );
+
+        // Exchange halo values
+        T_halo->gather( exec_space(), *T );
+    }
+
+    // Write the final state.
+    Cajita::Experimental::BovWriter::writeTimeStep( 0, 0, *T );
+}
+
+int main( int argc, char* argv[] )
+{
+    MPI_Init( &argc, &argv );
+    {
+        Kokkos::ScopeGuard scope_guard( argc, argv );
+
+        finiteDifference();
+    }
+    MPI_Finalize();
```

Comment:
```suggestion
    MPI_Init( &argc, &argv );
    Kokkos::initialize( argc, argv );

    finiteDifference();
    
    Kokkos::finalize();
    MPI_Finalize();
```
is more readable IMO

Reply:
Updated